docs/src/components/Header.astro

```diff
@@ -0,0 +1,334 @@
+---
+import { Icon } from 'astro-icon';
+---
+
+<div id="header">
+  <button id="sidebar-toggle"><Icon width="30px" name="mdi:menu" /></button>
+  <div id="theme-button-wrapper">
+    <Icon id="sun" width="30px" name="mdi:white-balance-sunny" />
+    <Icon id="moon" width="30px" name="mdi:weather-night" />
+    <button id="theme-button"></button>
+  </div>
+  <a href="/" id="header-text">The Eludris Developer Portal</a>
+  <a href="https://github.com/eludris/eludris">
+    <Icon name="akar-icons:github-fill" width="30px" />
+  </a>
+  <button id="search">
+    <Icon name="mdi:search" width="30px" />
+  </button>
+</div>
+<div id="search-bar" style="display: none;">
+  <input type="text" id="search-input" placeholder="Search For Something" />
+  <span id="search-stats"></span>
+  <div id="search-results"></div>
+</div>
+
+<script is:inline>
+  const sidebar = document.getElementById('sidebar');
+  const content = document.getElementById('content');
+
+  let theme =
+    localStorage.getItem('theme') ??
+    (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
```

Comment:
```suggestion
    (window.matchMedia.?('(prefers-color-scheme: dark)').matches
```

Reply:
I thought the optional chaining operator was `?.`, not `.?`

Reply:
^oops typo